node-notes-cli/app.js

```diff
@@ -0,0 +1,38 @@
+const fs = require('fs');
+const notebook = require('./data.json');
+
+const command = process.argv[2];
+const ID = notebook.nextId;
+const notes = notebook.notes;
+
+function updateNotebook() {
+  fs.writeFile('data.json', JSON.stringify(notebook, null, 2), err => {
+    if (err) throw err;
+  });
+}
+
+if (command === 'read') {
+  for (const property in notes) {
+    console.log(`${property}: ${notes[property]}`);
+  }
+} else if (command === 'create') {
+  notes[ID] = process.argv[3];
+  notebook.nextId++;
+
+  updateNotebook();
+} else if (command === 'update') {
+  for (const property in notes) {
+    if (property === process.argv[3]) {
+      notes[property] = process.argv[4];
+    }
+  }
```

Comment:
When you already know the property name you don't need a loop, you can just directly check if the property exists or not.

```JAVASCRIPT
const prop = process.argv[3];

if (notes[prop]) {
  // Update note
}
```

Reply:
When using if (notes[prop]), how does it find if it is truthy without looping and checking each of the properties in notes to see if it exists?

Reply:
If `prop` is `3` but notes doesn't have a property of `3` then `notes[3]` will be undefined which is falsey 

If `prop` is `5` and notes does have a property of `5` then `notes[5]` will be the value of the note which will be truthy.

You are just directly trying to access a specific property it is either there or it isn't